01_fib/main.go

```diff
@@ -4,16 +4,33 @@ import (
 	"fmt"
 	"io"
 	"os"
+	"strconv"
+	"strings"
 )
 
-var out io.Writer = os.Stdout
+var outWriter io.Writer = os.Stdout
+
+func calculateFib(n int) []int {
+	f0, f1, result := 1, 1, []int{}
+
+	for i := 1; i <= n; i++ {
+		result = append(result, f0)
+		f0, f1 = f1, f0+f1
+	}
+	return result
+}
 
 func fib(n int) {
-	a, b := 1, 1
-	for i := 0; i < n; i++ {
-		fmt.Fprintln(out, a)
-		a, b = b, a+b
+	//ideally this can be printed in main but lab needs to be printed inside the func
+	printResult(calculateFib(n))
+}
+
+func printResult(arrayToPrint []int) {
+	resultsAsStr := []string{}
```

Comment:
You should output directly to outWriter using `fmt.Sprintf()`.

Reply:
@anzdaddy The output is expected to have new lines, I am not sure if an int slice can be formatted to be printed with new lines